rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -649,6 +643,71 @@ impl PendingMessage {
         PendingOperationStatus::FirstPrepareAttempt
     }
 
+    /// A preflight check to see if a message could possibly meet
+    /// a gas payment requirement prior to undertaking expensive operations
+    /// like metadata building or gas estimation.
+    /// If the message does not meet the gas payment requirement,
+    /// Err(PendingOperationResult) is returned, with the PendingOperationResult intended
+    /// to be propagated up by the prepare fn.
+    async fn meets_gas_payment_requirement_preflight_check(
+        &mut self,
+    ) -> Result<(), PendingOperationResult> {
+        // We test if the message may meet the gas payment requirement
+        // with the most simple tx cost estimate: one that has zero cost
+        // whatsoever. If the message does not meet the gas payment requirement
+        // with zero cost, we can skip the metadata building and gas estimation
+        // altogether. This covers the case of a message that did not pay our IGP,
+        // which may violate the gas payment enforcement policies depending on
+        // the configuration, but also allows us to be tolerant of the configuration
+        // allowing no payment at all.
+        let zero_cost = TxCostEstimate {
+            gas_limit: U256::zero(),
+            gas_price: FixedPointNumber::zero(),
+            l2_gas_limit: None,
+        };
+
+        self.meets_gas_payment_requirement(&zero_cost)
```

Comment:
so rather than using `GasPaymentPolicyMinimum`, this reuses the configured IGP policy

Reply:
exactly, to still allow the relayer to relay messages without requiring any payments if it's configured to do so